src/org/etools/j1939_84/controllers/part09/Part09Step24Controller.java

```diff
@@ -54,12 +57,52 @@
     @Override
     protected void run() throws Throwable {
         // 6.9.24.1.a. DS DM33 [(send Request (PGN 59904) for PGN 41216 (SPNs 4124-4126))] to each OBD ECU.
-        // 6.9.24.2.a. Fail if any ECU reports a different number EI-AECD than was reported in part 2.
-        // [Engines using SI technology need not respond until the 2024 engine model year].
```

Comment:
The check needs to take into consideration the fuel type and model year.
The logic works out to be :
```
if (!sparkIgnition || getEngineModelYear() >= 2024) {
// do testing
}
```

Reply:
Does not say anything about that.

Reply:
Requirements are: 

6.9.24.2 Fail Criteria

a. Fail if any ECU reports a different number EI-AECD than was reported in part 2. [Engines using SI technology need not respond until the 2024 engine model year].

b. Compare to list of ECU address + EI-AECD number + actual time (for Timer 1 and/or Timer 2) for any with nonzero timer values created earlier in step 6.9.6.1 and fail if any timer value is less than the value it was earlier in this part.

c. Fail if NACK not received from OBD ECUs that did not provide a DM33 message.

Reply:
I do not see what you mean.

Reply:
For point C, we don't fail the engine is using SI and the model year is < 2024.